physionet-django/console/utility.py

```diff
@@ -31,6 +32,828 @@ class DOIExistsError(Exception):
+        project (project.models.Project): The project for which to delete files and the associated bucket.
+
+    Returns:
+        None
+
+    Note:
+    - The function leverages the 'delete_s3_bucket' method to perform the deletion.
+    - Ensure that AWS credentials (Access Key and Secret Key) with sufficient permissions
+      are properly configured for the S3 client used in this function.
+    """
+    if settings.AWS_PROFILE is not None:
+        boto3.setup_default_session(settings.AWS_PROFILE)
+        s3 = boto3.resource('s3')
+        bucket_name = get_bucket_name(project)
+        bucket = s3.Bucket(bucket_name)
+        # Empty the specified AWS S3 'bucket' by deleting all objects (files) within it
+        empty_s3_bucket(bucket)
+        # Delete the empty bucket itself
+        bucket.delete()
+
+
+def delete_list_of_projects_from_s3(projects):
+    """"
+    Bulk delete a list of projects from AWS S3.
+
+    This function deletes a list of projects, including their files and associated AWS S3 buckets.
+    If the list includes open projects, the 'delete_project_from_S3' function will be called only once
+    for the first open project encountered.
+
+    Args:
+        projects (list of project.models.Project): A list of projects to delete from AWS S3.
+
+    Returns:
+        None
+
+    Note:
+    - The function leverages the 'delete_project_from_S3' method to perform project deletions.
+    - Ensure that AWS credentials (Access Key and Secret Key) with sufficient permissions
+      are properly configured for the S3 client used in this function.
+    """
+    for project in projects:
+        delete_project_from_S3(project)
+        if project.access_policy == AccessPolicy.OPEN:
+            break
+
+def delete_all_projects_from_S3():
+    """
+    Bulk delete all published projects from AWS S3.
+
+    This function deletes all published projects, including their files and associated AWS S3 buckets.
+    It distinguishes between open and non-open projects, calling the 'delete_project_from_S3' function
+    accordingly. Non-open projects, such as 'RESTRICTED', 'CREDENTIALED', or 'CONTRIBUTOR_REVIEW', are
+    deleted individually. Open projects are stored in the same S3 bucket, so only one delete operation
+    is needed, which is performed by passing the first open project.
+
+    Args:
+        None
+
+    Returns:
+        None
+
+    Note:
+    - The function leverages the 'delete_project_from_S3' method to perform project deletions.
+    - Ensure that AWS credentials (Access Key and Secret Key) with sufficient permissions
+      are properly configured for the S3 client used in this function.
+    """
+    from project.models import PublishedProject, AccessPolicy
+    not_open_published_projects = PublishedProject.objects.filter(access_policy=AccessPolicy.RESTRICTED) | PublishedProject.objects.filter(access_policy=AccessPolicy.CREDENTIALED) | PublishedProject.objects.filter(access_policy=AccessPolicy.CONTRIBUTOR_REVIEW)
```

Comment:
this line is way over the maximum width of 119 characters. apply a formatter to reformat the code (be careful to apply the restyling to only your changes, not to existing code).